src/lib.rs

```diff
@@ -267,10 +228,8 @@ mod tests {
     fn basic_override_allow_all_extensions() {
         let treedir = "tests/fixtures/tree-basic";
         let dirs = vec![format!("{}/{}", treedir, "etc")];
-        let allowed_extensions = vec![];
-        let od_cfg = FragmentScanner::new(dirs, "liboverdrop.d", false, allowed_extensions);
 
-        let fragments = od_cfg.scan();
+        let fragments = scan::<_, _, _, &str>(&dirs, "liboverdrop.d", false, &[]);
```

Comment:
The need to turbofish so much is perhaps a bit of a warning sign of an overly generic API.

This may be a case where a builder API would be better.  I think most of the benefit of what you're going for here would be gained by having a struct which *borrows* its input, and keeping `scan()` as a method on it no?

Reply:
Disagree, it's a side-effect of passing an untyped `&[]` which never happens in practice; all examples *and* all users (accd'g to crates.io rdeps) pass either (a) `vec![String::from("conf")],` ([s/z-g](https://github.com/systemd/zram-generator/blob/1e15869bcff595b016820cbc0d94103998c60d65/src/config.rs#L6), [c/f-c-p](https://github.com/coreos/fedora-coreos-pinger/blob/d7fca197c9df75ff9cce9a35c540591f742f2f4d/src/config/inputs.rs#L26)) or (b) a concrete type from higher up ([c/z](https://github.com/coreos/zincati/blob/9c7294a85a6c410f69d183ae01db64fb3542b25e/src/config/inputs.rs#L27), and even then the only user literally passes `vec!["toml".to_string()];`).

You could just as well write this as `let extensions: &[&OsStr] = &[];` or whatever, but I don't expect anyone to ever actually run into this.